device/iothub_client_python/src/iothub_client_python.cpp

```diff
@@ -1513,6 +1514,17 @@ class IoTHubClient
         {
             throw IoTHubClientError(__func__, IOTHUB_CLIENT_ERROR);
         }
+        else
+        {
+            if (client_interface_type == CLIENT_INTERFACE_DEVICE)
+            {
+                IoTHubDeviceClient_SetOption(iotHubClientHandle, "product_info", PYTHON_PRODUCT_INFO);
```

Comment:
>"product_info" [](start = 65, length = 14)

Please use iothub_client_options.h, OPTION_PRODUCT_INFO (and in line below also).